src/charm.py

```diff
@@ -163,7 +201,7 @@ def _on_kafka_pebble_ready(self, event: PebbleReadyEvent) -> None:
             self.unit.status = BlockedStatus("kafka unit not connected to ZooKeeper")
             return
 
-    def _on_config_changed(self, event: ConfigChangedEvent) -> None:
+    def _on_config_changed(self, event: EventBase) -> None:
```

Comment:
On this method for the VM charm you added the following block of code:
```python
. . .
    if set(properties) ^ set(self.kafka_config.server_properties):
. . .
            if isinstance(event, StorageEvent):  # to get new storages
                self.snap.disable_enable("kafka")
                return
            else:
                self.on[f"{self.restart.name}"].acquire_lock.emit()
```
In this case it woudn't use snap commands, but should the Pebble equivalent be added here?

Reply:
Christ, VERY helpful catch, thank you!!

Reply:
For K8s, there isn't the ability to `add` a storage, we're limited to one, so I don't think we'd need this functionality.